lib/index.js

```diff
@@ -178,10 +185,10 @@ Socket.prototype._receive = function () {
             return;
         }
     }
-    if (msg == -1) return;
+  if (msg == -1) return '';
```

Comment:
Why do we change the function signature to return `null` vs `empty string`?  Why would have to differentiate between those two?  Both are failure conditions; we should continue to `return;` undefined.